osquery/tables/system/darwin/signature.mm

```diff
@@ -58,130 +62,159 @@ Status getVerifyFlags(SecCSFlags& flags) {
+
+  SecCSFlags flags = 0;
+  getVerifyFlags(flags);
+  result = SecStaticCodeCheckValidityWithErrors(
+      static_code, flags, nullptr, nullptr);
   if (result == errSecSuccess) {
-    isValidated = true;
+    r["signed"] = "1";
   } else {
     // If this errors, then we either don't have a signature, or it's malformed.
-    VLOG(1) << "Static code validity check failed for file: " << path;
+    r["signed"] = "0";
   }
-  CFDictionaryRef codeInfo = nullptr;
+
+  CFDictionaryRef code_info = nullptr;
   result = SecCodeCopySigningInformation(
-      staticCode, kSecCSSigningInformation | kSecCSRequirementInformation,
-      &codeInfo);
-  if (result == errSecSuccess) {
-    // If we don't get an identifier for this file, then it's not signed.
-    CFStringRef ident =
-        (CFStringRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoIdentifier);
-    if (ident != nullptr) {
-      // We have an identifier - this indicates that the file is signed,
-      // and, since it didn't error above, it's *also* a valid signature.
-      if (isValidated) {
-        r["signed"] = INTEGER(1);
-      }
-      r["identifier"] = stringFromCFString(ident);
+      static_code,
+      kSecCSSigningInformation | kSecCSRequirementInformation,
+      &code_info);
+
+  if (result != errSecSuccess) {
+    results.push_back(r);
+    CFRelease(static_code);
+    return Status(1, "Could not get signing information for file");
+  }
 
-      // Get CDHash
-      r["cdhash"] = "";
-      CFDataRef hashInfo =
-          (CFDataRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
-      if (hashInfo != nullptr) {
-        r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
-        // Get the SHA-1 bytes
-        std::stringstream ss;
-        auto bytes = CFDataGetBytePtr(hashInfo);
-        if (bytes != nullptr &&
-            CFDataGetLength(hashInfo) == CC_SHA1_DIGEST_LENGTH) {
-          // Write bytes as hex strings
-          for (size_t n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
-            ss << std::hex << std::setfill('0') << std::setw(2);
-            ss << (unsigned int)bytes[n];
-          }
-          r["cdhash"] = ss.str();
-        }
-        if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
-          VLOG(1) << "Error extracting code directory hash";
-          r["cdhash"] = "";
-        }
-      }
+  // If we don't get an identifier for this file, then it's not signed.
+  CFStringRef ident =
+      (CFStringRef)CFDictionaryGetValue(code_info, kSecCodeInfoIdentifier);
 
-      // Team Identifier
-      r["team_identifier"] = "";
-      CFTypeRef teamIdent = nullptr;
-      if (CFDictionaryGetValueIfPresent(codeInfo, kSecCodeInfoTeamIdentifier,
-                                        &teamIdent)) {
-        r["team_identifier"] = stringFromCFString((CFStringRef)teamIdent);
+  if (ident == nullptr) {
+    CFRelease(code_info);
+    results.push_back(r);
+    CFRelease(static_code);
+    return Status(1, "No identifier found for file(" + arch + "): " + path);
```

Comment:
I feel like this error is not consistent with the above. I'd recommend just reporting that the identifier is not found for the arch.